src/character.cpp

```diff
@@ -3011,6 +3016,30 @@ bool Character::practice( const skill_id &id, int amount, int cap, bool suppress
     return level_up;
 }
 
+// The dumbest shit in the universe. This code exists to correct a deficiency in
+// the existing night vision code.
+// Normal Vision = 1.374320
+// Night Vision = 1.272479
+// High Night Vision = 1.155692
+// Full Night Vision = .934998
+// We want high/full NV to enable crafting in the dark, but they don't scale linearly, so here we are
+float Character::interpolate_night_vision( float vision_threshold ) const
```

Comment:
You should use [lerp](https://github.com/Cataclysm-TLG/Cataclysm-TLG/blob/e02b1f4b36956bfdfc09c419313526e11acfaf8d/src/cata_utility.h#L231) or [multi-lerp](https://github.com/Cataclysm-TLG/Cataclysm-TLG/blob/master/src/cata_utility.cpp#L267)

Reply:
I have used Lerp a ton in the past, and somehow completely forgot it existed.